tests/zfs-tests/tests/functional/persist_l2arc/persist_l2arc.kshlib

```diff
@@ -0,0 +1,57 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+function cleanup
+{
+	if datasetexists $TESTPOOL ; then
+		log_must zpool destroy -f $TESTPOOL
+	fi
+}
+
+#
+# Set l2arc_noprefetch tunable to $value
+#
+function sysctl_l2arc_noprefetch # value
+{
+	typeset value="$1"
+
+	if is_linux; then
+		log_note "Setting l2arch_noprefetch tunable to $value"
+		log_must eval "echo $value > "\
+		    "/sys/module/zfs/parameters/l2arc_noprefetch"
+	fi
+}
+
+#
+# Set l2arc_rebuild_enabled tunable to $value
+#
+function sysctl_l2arc_rebuild_enabled # value
+{
+	typeset value="$1"
+
+	if is_linux; then
+		log_note "Setting l2arch_rebuild_enabled tunable to $value"
+		log_must eval "echo $value > "\
+		    "/sys/module/zfs/parameters/l2arc_rebuild_enabled"
+	fi
+}
```

Comment:
I'm sorry, I should I have mentioned this in my original comment.  The `set_tunable32` function already exists for this purpose and includes the needed compatibility code.  As an example, I'd suggest taking a look at `trim/autotrim_config.ksh`.  What we'd like to do is preserve the original value, change, it and then restore it on exit in the cleanup handler.